### PR TITLE
Gpu stream thread safety

### PR DESCRIPTION
**Pull request by WeiqunZhang**

### CODE REVIEW

File: Src/Base/AMReX_OpenMP.H [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#ifdef _OPENMP
+#include <omp.h>
+
+namespace amrex {
+namespace OpenMP {
+
+    inline int get_num_threads () { return omp_get_num_threads(); }
+    inline int get_max_threads () { return omp_get_max_threads(); }
+    inline int get_thread_num  () { return omp_get_thread_num();  }
+    inline int in_parallel     () { return omp_in_prallel();      }
```
Comment (kngott) [inline Src/Base/AMReX_OpenMP.H]: `omp_in_parallel` is spelled incorrectly.

Comment (maximumcats) [inline Src/Base/AMReX_OpenMP.H]: Typo here

Comment (WeiqunZhang) [reply Src/Base/AMReX_OpenMP.H]: Good catch.  Fixed now.

**Review by kngott — APPROVED** [review]

File: Src/Base/AMReX_OpenMP.H (hunk 1 above)
Comment (WeiqunZhang) [reply Src/Base/AMReX_OpenMP.H]: I think that was fixed already.

Comment (kngott) [reply Src/Base/AMReX_OpenMP.H]: Yeah. I believe I made this around the same time as Max. I'm not sure why, but my approval seemed to reannounce this comment.

**Review by maximumcats — APPROVED** [review]

**Pull request merged**